File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/lookuptable/NormAbbreviation.java
package de.bund.digitalservice.ris.caselaw.domain.lookuptable;

import de.bund.digitalservice.ris.caselaw.domain.lookuptable.documenttype.DocumentType;
import java.time.Instant;
import java.util.List;
import java.util.UUID;
import lombok.Builder;

@Builder
public record NormAbbreviation(
    UUID id,
    String abbreviation,
    Instant decisionDate,
    Long documentId,
    String documentNumber,
    String officialLetterAbbreviation,
    String officialLongTitle,
    String officialShortTitle,
    String source,
    List<DocumentType> documentTypes,
    Region region) {}

File: frontend/src/domain/normAbbreviation.ts
export type NormAbbreviation = {
  id?: string
  abbreviation: string
  decisionDate?: string
  documentId?: string
  documentNumber?: string
  officialLetterAbbreviation?: string
  officialLongTitle?: string
  officialShortTitle?: string
  source?: string
  documentTypes?: string[]
  regions?: Region[]
}

export type Region = {
  code: string
}
